### DISCUSSION TITLE
barLine@barplace and barLine@tstamp questions

### DISCUSSION THREAD

**Discussion by craigsapp** [music-encoding/music-encoding#820] [Q&A]
In the following example, how should the second barlines between the two parts be aligned:

<img width="797" alt="screen shot 2018-10-04 at 8 24 26 am" src="https://user-images.githubusercontent.com/3487289/46484650-044a9100-c7af-11e8-8574-a2d7cd63e949.png">

In other words, the E4 note on the top staff should be split in half for the placement of the barline.

In theory this looks like it should be done with `@tstamp`.  But what is the reference point for the timestamp in such a case?  Is it from the start of the music (or start of the `<measure>` which contains the note,) or from the last barline present in the music?

The intention of `@barplace="mensur"` is to display the barlines in the Mensurstrich style:
      https://en.wikipedia.org/wiki/Mensurstrich
This is, of course, not implemented [yet] in verovio.  But if it were to be implemented, how would the placement of the barline be done?  Would there only need to be one `@barplace` for all Mensurstrich occuring at the same time?  If not, then where would the barline be placed?: above or below the current staff, or both above and below the current staff?

**Comment by bwbohl** [comment]
could you please provide the corresponding XML?

**Comment by craigsapp** [comment]
Here is an approximate MEI encoding:

```xml
<?xml version="1.0" encoding="UTF-8"?>
<?xml-model href="http://music-encoding.org/schema/4.0.0/mei-all.rng" type="application/xml" schematypens="http://relaxng.org/ns/structure/1.0"?>
<?xml-model href="http://music-encoding.org/schema/4.0.0/mei-all.rng" type="application/xml" schematypens="http://purl.oclc.org/dsdl/schematron"?>
<mei xmlns="http://www.music-encoding.org/ns/mei" meiversion="4.0.0">
 <meiHead>
  <fileDesc>
   <titleStmt>
    <title />
   </titleStmt>
   <pubStmt />
  </fileDesc>
  <encodingDesc>
   <appInfo>
    <application isodate="2019-04-04T11:22:34" version="2.1.0-dev-5f615b1">
     <name>Verovio</name>
     <p>Transcoded from Humdrum</p>
    </application>
   </appInfo>
  </encodingDesc>
  <workList>
   <work>
    <title />
   </work>
  </workList>
 </meiHead>
 <music>
  <body>
   <mdiv xml:id="mdiv-0000001133745031">
    <score xml:id="score-0000000230336186">
     <scoreDef xml:id="scoredef-0000001230011979" midi.bpm="300">
      <staffGrp xml:id="staffgrp-0000001536102058" symbol="brace" bar.thru="true">
       <staffDef xml:id="staffdef-0000000935815250" clef.shape="G" clef.line="2" mensur.sign="O" meter.unit="1" n="1" lines="5"/>
       <staffDef xml:id="staffdef-0000000935815250" clef.shape="G" clef.line="2" mensur.sign="O" meter.unit="1" n="2" lines="5"/>
      </staffGrp>
     </scoreDef>
     <section xml:id="section-L1F1">
       <staff xml:id="staff-0000000564622735" n="1">
        <layer xml:id="layer-L1F1N1" n="1">
         <note xml:id="note-L5F1" dots="1" dur="breve" oct="4" pname="g" accid.ges="n" />
         <barLine barplace="mensur"/>
         <note xml:id="note-L7F1" dur="breve" oct="4" pname="f" accid.ges="n" />
         <note xml:id="note-L8F1" dur="breve" oct="4" pname="e" accid.ges="n" />
         <barLine barplace="mensur"/>
         <note xml:id="note-L10F1" dur="breve" oct="4" pname="c" accid.ges="n" />
         <barLine barplace="mensur"/>
         <note xml:id="note-L12F1" dots="1" dur="breve" oct="4" pname="d" accid.ges="n" />
         <barLine/>
        </layer>
       </staff>
       <staff xml:id="staff-L5F2N1" n="2">
        <layer xml:id="layer-L1F2N1" n="1">
         <note xml:id="note-L6F2" dur="1" oct="4" pname="c" accid.ges="n" />
         <note xml:id="note-L7F2" dur="breve" oct="4" pname="d" accid.ges="n" />
         <barLine barplace="mensur"/>
         <note xml:id="note-L9F2" dots="1" dur="breve" oct="4" pname="e" accid.ges="n" />
         <barLine barplace="mensur"/>
         <note xml:id="note-L11F2" dur="breve" oct="4" pname="f" accid.ges="n" />
         <note xml:id="note-L12F2" dur="1" oct="4" pname="g" accid.ges="n" />
         <barLine barplace="mensur"/>
         <note xml:id="note-L14F2" dots="1" dur="breve" oct="4" pname="d" accid.ges="n" />
         <barLine/>
        </layer>
       </staff>
      </measure>
     </section>
    </score>
   </mdiv>
  </body>
 </music>
</mei>
```

The current rendering of this MEI data in verovio:

<img width="1034" alt="Screen Shot 2019-04-04 at 11 51 58 AM" src="https://user-images.githubusercontent.com/3487289/55569437-93d5eb80-56cf-11e9-8ee8-4fe9c6085f34.png">

The final target rendering mit Mensurstrich:

<img width="924" alt="Screen Shot 2019-04-04 at 11 31 06 AM" src="https://user-images.githubusercontent.com/3487289/55569511-b10aba00-56cf-11e9-86d3-6d33451e15aa.png">

The problem barline is after the E in the top voice.  If barred in a modern style, this would be the rendering:

<img width="905" alt="Screen Shot 2019-04-04 at 11 58 48 AM" src="https://user-images.githubusercontent.com/3487289/55569849-47d77680-56d0-11e9-8247-d18f8ea33656.png">
